packages/zero-runtime/src/styled.d.ts

```diff
@@ -42,55 +24,63 @@ export type StyledCallback<Props extends {}> = (buildArg: ThemeArgs) => StyledCs
 
 export type StyledArgument<Props extends {}> = StyledCssArgument<Props> | StyledCallback<Props>;
 
-export interface StyledComponent<
-  ComponentProps extends {},
-  SpecificComponentProps extends {} = {},
-  JSXProps extends {} = {},
-> extends React.FC<ComponentProps & SpecificComponentProps & JSXProps> {}
+export type PolymorphicComponentProps<
+  BaseProps extends object,
+  AsTarget extends React.ElementType | undefined,
+  AsTargetProps extends object = AsTarget extends React.ElementType
+    ? React.ComponentPropsWithRef<AsTarget>
+    : {},
+> = NoInfer<Omit<Substitute<BaseProps, AsTargetProps>, 'as'>> & {
+  as?: AsTarget;
+  sx?: SxProp;
+};
+
+export interface PolymorphicComponent<BaseProps extends object>
+  extends React.ForwardRefExoticComponent<BaseProps> {
+  <AsTarget extends React.ElementType | undefined = undefined>(
+    props: PolymorphicComponentProps<BaseProps, AsTarget>,
+  ): JSX.Element;
+}
+
+export interface StyledComponent<Props extends object = {}> extends PolymorphicComponent<Props> {
+  defaultProps?: Partial<Props> | undefined;
+  toString: () => string;
+}
 
-export interface CreateStyledComponent<
-  ComponentProps extends {},
-  SpecificComponentProps extends {} = {},
-  JSXProps extends {} = {},
-> {
+export interface CreateStyledComponent<Component extends React.ElementType, OuterProps extends {}> {
   /**
-   * @typeparam AdditionalProps  Additional props to add to your styled component
+   * @typeparam Props: Additional props to add to the styled component
    */
-  <AdditionalProps extends {} = {}>(
-    ...styles: Array<StyledArgument<ComponentProps & SpecificComponentProps & AdditionalProps>>
-  ): StyledComponent<ComponentProps & AdditionalProps, SpecificComponentProps, JSXProps>;
+  <Props extends {} = {}>(...styles: Array<StyledArgument<OuterProps & Props>>): StyledComponent<
```

Comment:
```suggestion
  <Props extends Record<string, any> = {}>(...styles: Array<StyledArgument<OuterProps & Props>>): StyledComponent<
```
`{}` could be number, string, etc. Would this work?

Reply:
It'll work technically but then users won't be able to pass anything. So ideally they won't use it like that. It's for

```ts
const A = styled.a<{isActive: boolean}>({
});
```

I doubt users would set it to string/number/primitive. WDYT. Not a fan of `any` but it could be `Record<string, unknown>` but I am not sure what the implications of that are.

Reply:
Got it, `Record<string, unknown>` sounds better. I think it's better than `{}` to force Props to be key-value map.